File: database/src/main/java/org/difft/app/database/models/SpeechToTextModel.java
package org.difft.app.database.models;

import com.tencent.wcdb.WCDBField;
import com.tencent.wcdb.WCDBIndex;
import com.tencent.wcdb.WCDBTableCoding;
import java.util.Objects;


@WCDBTableCoding
public class SpeechToTextModel {
    @WCDBField(isPrimary = true, isAutoIncrement = true)
    public long databaseId;

    @WCDBIndex(isUnique = true)
    @WCDBField
    public String messageId;

    /**
     *     Invisible(0),
     *     Converting(1),
     *     Show(2),
     */
    @WCDBField
    public int convertStatus;

    @WCDBField
    public String speechToTextContent;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        SpeechToTextModel that = (SpeechToTextModel) o;
        return databaseId == that.databaseId && convertStatus == that.convertStatus && Objects.equals(messageId, that.messageId) && Objects.equals(speechToTextContent, that.speechToTextContent);
    }

    @Override
    public int hashCode() {
        return Objects.hash(databaseId, messageId, convertStatus, speechToTextContent);
    }
}

File: database/src/main/java/org/difft/app/database/models/SpeechToTextModel.kt
package org.difft.app.database.models

import com.tencent.wcdb.WCDBField
import com.tencent.wcdb.WCDBIndex
import com.tencent.wcdb.WCDBTableCoding
import java.util.Objects

@WCDBTableCoding
class SpeechToTextModel {
    // Int (not Long): WCDB-KSP generates `== 0` autoincrement guard which does not
    // compile for Long primary keys. Column type stays ColumnType.Integer — no schema
    // change. Local rowid never exceeds 2^31 and is never externally serialized. #901
    @WCDBField(isPrimary = true, isAutoIncrement = true)
    var databaseId: Int = 0

    @WCDBIndex(isUnique = true)
    @WCDBField
    var messageId: String? = null

    /**
     *     Invisible(0),
     *     Converting(1),
     *     Show(2),
     */
    @WCDBField
    var convertStatus: Int = 0

    @WCDBField
    var speechToTextContent: String? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        other as SpeechToTextModel
        // exclude databaseId: rowid must not affect content equality, #901
        return convertStatus == other.convertStatus &&
                messageId == other.messageId &&
                speechToTextContent == other.speechToTextContent
    }

    // exclude databaseId: rowid must not affect content equality, #901
    override fun hashCode(): Int = Objects.hash(messageId, convertStatus, speechToTextContent)
}
